enterprise/cypher/physical-planning/src/main/scala/org/neo4j/cypher/internal/compatibility/v3_4/runtime/SlotConfiguration.scala

```diff
@@ -159,25 +68,24 @@ class PipelineInformation(private val slots: mutable.Map[String, Slot],
   def isAlias(key: String): Boolean = {
     aliases.contains(key)
   }
-  private val slotAliases = new mutable.HashMap[Slot, mutable.Set[String]] with mutable.MultiMap[Slot, String]
 
   def apply(key: String): Slot = slots.apply(key)
 
   def get(key: String): Option[Slot] = slots.get(key)
 
-  def add(key: String, slotInformation: Slot): Unit = slotInformation match {
+  def add(key: String, slot: Slot): Unit = slot match {
     case LongSlot(_, nullable, typ) => newLong(key, nullable, typ)
     case RefSlot(_, nullable, typ) => newReference(key, nullable, typ)
   }
 
-  def breakPipelineAndClone(): PipelineInformation = {
-    val newPipeline = new PipelineInformation(this.slots.clone(), numberOfLongs, numberOfReferences)
+  def copy(): SlotConfiguration = {
+    val newPipeline = new SlotConfiguration(this.slots.clone(), numberOfLongs, numberOfReferences)
```

Comment:
Maybe would be nicer to have alias and slotAlias part of constructor,
something like

```
class SlotConfiguration(private val slots: mutable.Map[String, Slot],
                          val initialNumberOfLongs: Int,
                          val initialNumberOfReferences: Int,
                          aliases: mutable.Set[String] = mutable.Set(),
                          slotAliases: SlotMap = new SlotMap
                ) 
```
where
```
type SlotMap =  mutable.HashMap[Slot, mutable.Set[String]] with mutable.MultiMap[Slot, String]
```

Reply:
I find this change attractive, but wasn't very happy with the result when I tried to do it. Maybe we can change that in a next step? I get ugliness when cloning the `SlotMap` type.